### ISSUE TITLE
Unexpected behavior of defaultTags

### ISSUE THREAD

**Issue by estiller** [pulumi/pulumi-awsx#1059]
### What happened?

The `@pulumi/aws` provider recently fixed its handling of `defaultTags`. When a user changes the `defaultTags` configuration with newer provider versions, these tags are applied to existing and updated resources. This is covered in https://github.com/pulumi/pulumi-aws/issues/1655.

However, I still get the old behaviour when using `@pulumi/awsx`. In the same TypeScript stack, I have resources from both `@pulumi/aws` and `@pulumi/awsx`, and when I update the `defaultTags`, they are not applied to the `awsx` resources.

This is an inconsistency between the providers and a problem when updating a tag across many resources in an existing stack. Also see https://github.com/pulumi/pulumi-aws/issues/1655#issuecomment-1633474264.

### Expected Behavior

The new `defaultTags` should also apply to existing `@pulumi/awsx` resources.

### Steps to reproduce

Here are steps to reproduce:

1. Create a TypeScript stack with a resource from `@pulumi/awsx`.
2. Update to the latest versions of `@pulumi/aws` and `@pulumi/awsx`.
3. Deploy it.
4. Change the `defaultTags`.
5. Deploy the stack again.

The result is that the `defaultTags` are not updated for the `awsx` resources.

### Output of `pulumi about`

```
CLI          
Version      3.76.1
Go Version   go1.20.6
Go Compiler  gc

Plugins
NAME    VERSION
aws     5.42.0
awsx    1.0.2
docker  3.6.1
nodejs  unknown
random  4.13.2

Host     
OS       darwin
Version  13.4.1
Arch     arm64

This project is written in nodejs: executable='/Users/.../.nvm/versions/node/v19.0.0/bin/node' version='v19.0.0'

Dependencies:
NAME            VERSION
@pulumi/aws     5.42.0
@pulumi/awsx    1.0.2
@pulumi/pulumi  3.75.0
@pulumi/random  4.13.2
@types/node     16.18.38
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by danielrbradley** [comment]
Hi @estiller thanks for reaching out.

This fixed default tags behaviour will be available in AWSx very soon. AWSx uses its own version of the AWS provider internally - and doesn't just use the plugin version installed by the user. 

There's going to be a new major version of the AWS provider released soon (v6.0.0) and we will also be updating the dependencies in AWSx to v6 after that release.

**Comment by mnlumi** [comment]
This is resolved in with the recent release of AWSx 2.0. I verified using awsx 2.1.1.
